centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -198,4 +198,71 @@ private function hasRestrictedAccessToHostCategories(array $accessGroupIds): boo
 
         return (bool) $statement->fetchColumn();
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists(int $hostCategoryId): bool
+    {
+        $this->info('Check existance of host category with id #' . $hostCategoryId);
+
+        $request = $this->translateDbName(
+            'SELECT hc.hc_id FROM `:db`.hostcategories hc WHERE hc.hc_id = :hostCategoryId'
+        );
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':hostCategoryId', $hostCategoryId, \PDO::PARAM_INT);
+        $statement->execute();
+
+        return (bool) $statement->fetchColumn();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function existsByAccessGroups(int $hostCategoryId, array $accessGroups): bool
+    {
+        $this->info(
+            'Check existance of host category by access groups',
+            ['id' => $hostCategoryId, 'accessgroups' => $accessGroups]
+        );
+
+        if (empty($accessGroups)) {
+            $this->debug('Access groups array empty');
+            return false;
+        }
+
+        $concat = new SqlConcatenator();
+
+        $accessGroupIds = array_map(
+            fn($accessGroup) => $accessGroup->getId(),
```

Comment:
Please check the code style on this file 🙏🏻 with php-cs-fixer
```suggestion
            fn ($accessGroup) => $accessGroup->getId(),
```